operator: dsa: Add provisioning configurability

ozhuraki opened the pull request:
The provisioning config can be optionally stored in the ProvisioningConfig configMap which is then passed to initcontainer through the volume mount.

There's also a possibility for a node specific congfiguration through passing a nodename via NODE_NAME into initcontainer's environment and passing a node specific profile via configMap volume mount.

Closes: #704

bart0sh reviewed (dismissed):
/lgtm

ozhuraki commented:
@bart0sh 
Changes: Minor fixes in doc, mainly to re-trigger the CI (failing on pull in gpu image).

mythi commented on deployments/dsa_plugin/overlays/dsa_initcontainer/dsa_initcontainer.yaml:
```diff
@@ -5,15 +5,26 @@ metadata:
 spec:
   template:
     spec:
+      provisioningConfig: intel-dsa-config
```
`provisioningConfig` does not exist in PodSpec

mythi commented on pkg/controllers/dsa/controller.go:
```diff
@@ -225,6 +235,25 @@ func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 			},
 		})
+
+		if devicePlugin.Spec.ProvisioningConfig != "" {
```
let's add checks in the validating webhook to warn users when user sets `ProvisioningConfig` but `InitImage` is empty

ozhuraki commented:
Changes:
- Removed provisioningConfig from the PodSpec in the sample deployment
- Added a check in in the validating webhook for ProvisioningConfig set but no InitImage
- Renamed dsa initcontainer image to idxd

mythi reviewed (approved):
LGTM

bart0sh reviewed (approved):
/lgtm
